components/gfx/text/text_run.rs

```diff
@@ -304,6 +304,20 @@ impl<'a> TextRun {
         })
     }
 
+    /// Returns the index in the range of the first glyph advancing over given advance
+    pub fn range_index_of_advance(&self, range: &Range<ByteIndex>, advance: Au) -> usize {
+        // TODO(Issue #199): alter advance direction for RTL
+        // TODO(Issue #98): using inter-char and inter-word spacing settings when measuring text
+        let mut remaining = advance;
+        self.natural_word_slices_in_range(range)
+            .fold(0, |index, slice| {
+                let (new_index, new_remaining) =
+                    slice.glyphs.range_index_of_advance(&slice.range, remaining, self.extra_word_spacing);
+                remaining = new_remaining;
+                new_index + index
```

Comment:
I guess you mean doing a `map` and `sum` which would indeed look nicer. That's just a habit from my early haskell days.